File: leetcode/3736.rs
impl Solution {
    pub fn min_moves(nums: Vec<i32>) -> i32 {
        let nums_max: &i32 = nums.iter().max().expect("non-empty vector");

        nums.iter().fold(0, |moves, n|
            moves + nums_max - n
        )
    }
}

File: leetcode/3736.cpp
class Solution {
  public:
    int minMoves(vector<int> &nums) {
        const int max_n = std::ranges::fold_left(
            nums, 0, [](int a, int b) { return std::max(a, b); });

        const int res = std::ranges::fold_left(
            nums, 0, [max_n](int acc, int n) { return acc + max_n - n; });

        return res;
    }
};
